lib/src/visitor/async_evaluate.dart

```diff
@@ -1601,6 +1602,9 @@ class _EvaluateVisitor
       _importSpan = span;
 
       var importCache = _importCache;
+      if (isBrowser && importCache == null && _importer == null) {
+        throw "Custom importers are required to `@use` or `@import` when compiling in the browser.";
+      }
```

Comment:
@nex3 I think I got a bit carried away in https://github.com/oddbird/dart-sass/commit/7feffba46b25157c293cd18891650cccba47cc0b, making sure that both `importCache` and `_importer` would be null here. I think that's required for `_importer`, but wouldn't be necessary if there's a way to check whether `importCache` has any `_importers` defined? I'd be happy for your feedback on this. 🙏

Reply:
> From a design perspective, it might be better to having a default `BrowserImporter` that literally just throw an error message when import is attempted. Use it to replace `NoOpImporter` or `FileSystemImporter` under browser environment. - This way you can avoid the need of making `importer` parameter `nullable` and perhaps save a few additional `null` checks.

@ntkme What do you think about this approach? https://github.com/sass/dart-sass/pull/1895/files/49182bdddad87cb7ca5d14862ea76a028d4e52e6..18d9c985b9823e25f0e21bd041d16398db73dc3e

I retained `NoOpImporter` for the base importer, but then added a default `BrowserImporter` that throws an error if no other `importers` are defined. That way users can provide a custom `importer` or `importers` if they want, but if both are missing then it will throw an error. One downside I see is that if a user-supplied `importer` does not find the imported stylesheet and the request falls through to `importers`, I think it will now throw the "Custom importers are required" error instead of a more helpful "Can't find stylesheet to import" error.

Reply:
I'm not that familiar with logic around this, maybe something like `nodeImporter ? NoOpImporter() : BrowserImporter()`? E.g. If user defined an importer, you can use `NoOp` so that it does not fall-over and throw, if user did not define the importer, we pick our `BrowserImporter` to throw.

Reply:
Cleaned up a bit and resolved the mentioned issue: https://github.com/sass/dart-sass/pull/1895/files/49182bdddad87cb7ca5d14862ea76a028d4e52e6..aa3afabda16fe95cb899fdef98c3308811318069

@nex3 @ntkme What do you think of this solution?

Reply:
Much simpler. LGTM

Reply:
Looks good to me as well!